app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java

```diff
@@ -151,52 +155,139 @@ public static void e(String a, String b){
         }
-        public static void wtf(String a, String b, Exception e){
-            android.util.Log.wtf(a, b, e);
-            UserError.UserErrorHigh(a, b + "\n" + e.toString());
+        public static void wtf(String tag, String b, Exception e){
+            android.util.Log.wtf(tag, b, e);
+            UserError.UserErrorHigh(tag, b + "\n" + e.toString());
         }
-        public static void wtf(String a, Exception e){
-            android.util.Log.wtf(a, e);
-            UserError.UserErrorHigh(a, e.toString());
+        public static void wtf(String tag, Exception e){
+            android.util.Log.wtf(tag, e);
+            UserError.UserErrorHigh(tag, e.toString());
         }
 
-        public static void uel(String a, String b) {
-            android.util.Log.i(a, b);
-            UserError.UserEventLow(a, b);
+        public static void uel(String tag, String b) {
+            android.util.Log.i(tag, b);
+            UserError.UserEventLow(tag, b);
         }
 
-        public static void ueh(String a, String b) {
-            android.util.Log.i(a, b);
-            UserError.UserEventHigh(a, b);
+        public static void ueh(String tag, String b) {
+            android.util.Log.i(tag, b);
+            UserError.UserEventHigh(tag, b);
         }
 
-        public static void d(String a, String b){
-            android.util.Log.d(a, b);
+        public static void d(String tag, String b){
+            android.util.Log.d(tag, b);
+        	if(ExtraLogTags.shouldLogTag(tag, android.util.Log.DEBUG)) {
+        		UserErrorLow(tag, b);
+        	}
         }
 
-        public static void v(String a, String b){
-            android.util.Log.v(a, b);
+        public static void v(String tag, String b){
+            android.util.Log.v(tag, b);
+        	if(ExtraLogTags.shouldLogTag(tag, android.util.Log.VERBOSE)) {
+        		UserErrorLow(tag, b);
+        	}           
         }
 
-        public static void i(String a, String b){
-            android.util.Log.i(a, b);
+        public static void i(String tag, String b){
+            android.util.Log.i(tag, b);
+        	if(ExtraLogTags.shouldLogTag(tag, android.util.Log.INFO)) {
+        		UserErrorLow(tag, b);
+        	}
         }
+        
+        static ExtraLogTags extraLogTags = new ExtraLogTags();
+    }
+    
+    public static class ExtraLogTags {
+    	
+    	
+    	static Hashtable <String, Integer> extraTags;
+    	ExtraLogTags () {
+    		extraTags = new Hashtable <String, Integer>();
+    		String extraLogs = Home.getPreferencesStringDefaultBlank("extra_tags_for_logging");
+    		readPreference(extraLogs);
+    	}
+    	
+    	/*
+    	 * This function reads a string representing tags that the user wants to log
+    	 * Format of string is tag1:level1,tag2,level2
+    	 * Example of string is Alerts:i,BG:W
+    	 * 
+    	 */
+    	public static void readPreference(String extraLogs) {
+    		Log.e("dddd", "called " +  extraLogs);
```

Comment:
This and similar lines, is `dddd` just an internal tag for your use when debugging this code or does it have any special significance?